src/Data/PreciseDateTime.purs

```diff
@@ -28,7 +29,7 @@ import Data.RFC3339String (RFC3339String(..), trim)
 import Data.RFC3339String as RFC3339String
 import Data.String (Pattern(Pattern), drop, length, split, takeWhile)
 import Data.Time.Duration as Duration
-import Data.Time.PreciseDuration (PreciseDuration, toMilliseconds, toNanoseconds, unPreciseDuration)
+import Data.Time.PreciseDuration (PreciseDuration(..), toMilliseconds, toNanoseconds, unPreciseDuration)
 
 data PreciseDateTime = PreciseDateTime DateTime Nanosecond
```

Comment:
Something I noted to @paulyoung  is that using a BigInt for Nanosecond here is unnecessary. Since this only represents a subsecond value, a regular `Int` should be fine. Currently the representation supports nanosecond values greater than or equal to 1 second, which is weird.

Reply:
I believe this is actually an `Int` since it's `Nanosecond` from `Data.PreciseDateTime.Component`, which is also `Bounded` to be less than 1 second.